src/java/main/org/apache/zookeeper/Login.java

```diff
@@ -197,10 +206,7 @@ public void run() {
                         break;
                     }
                     if (isUsingTicketCache) {
-                        String cmd = "/usr/bin/kinit";
-                        if (System.getProperty("zookeeper.kinit") != null) {
-                            cmd = System.getProperty("zookeeper.kinit");
-                        }
+                        String cmd = zkConfig.getProperty(ZKConfig.KINIT_COMMAND, KINIT_COMMAND_DEFAULT);
```

Comment:
I would create a ``zkConfig#getKinitCommand()`` method here to encapsulate this ``getProperty`` call (that OO concepts separation of concerns, data hiding etc), but **it's up to you** (REALLY!). I am fine as with current version.